backend/src/controllers/profileController.ts

```diff
@@ -0,0 +1,66 @@
+import { v4 } from "uuid";
+import { createProfileSchema, getProfileSchema } from "../helpers/schemas";
+import dbPool from "../db/index";
+import { NextFunction, Request, Response } from "express";
+
+interface NewUser {
+  id: string;
+  name: string;
+  email: string;
+  hobby: string;
+}
+
+export default class ProfileController {
+  static async getProfiles(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<void> {
+    try {
+      const pool = dbPool.getPool();
```

Comment:
It's still very tied to the specific pool implementation, so this class should take it as a constructor parameter or as a function argument ( and it has to be an abstract interface, not a concrete implementation)